kubectl-rook-ceph.sh

```diff
@@ -273,6 +276,97 @@ function run_purge_osd() {
     rook ceph osd remove --osd-ids=$1 --force-osd-removal=$force_removal"
 }
 
+####################################################################################################
+# 'kubectl rook-ceph debug commands
+####################################################################################################
+
+function update_deployment_spec_command(){
+  deployment_spec=$1
+  main_command='["sleep","infinity"]'
+  deployment_spec=$(jq .template.spec.containers[0].command="$main_command" <<< $deployment_spec)
+  deployment_spec=$(jq .template.spec.containers[0].args=[] <<< $deployment_spec)
+  echo "$deployment_spec" # for returning the updated deployment_spec
+}  
+
+function update_deployment_spec_image(){
+  deployment_spec=$1
+  alternalte_image=\"$2\"
+  deployment_spec=$(jq .template.spec.containers[0].image="$alternalte_image" <<< $deployment_spec)
+  echo "$deployment_spec" # for returning the updated deployment_spec
+}
+
+function verify_deployment_exist(){
+  if [[ $deployment_name != *"mon"* ]] && [[ $deployment_name != *"osd"* ]]; then
+    fail_error "only mon or osd deployment name can passed for starting debug mode"
+  fi
+  if _="$(KUBECTL_NS_CLUSTER get deployment "$deployment_name")"; then
+    echo "setting debug mode for \"$deployment_name\"" 
+  else  
+    fail_error "deployment $deployment_name doesn't exist"
+  fi
+}
+
+function run_start_debug(){
+  [[ -z "${1:-""}" ]] && fail_error "Missing mon or osd deployment name"
+  deployment_name="$1"
+  alternate_image_flag="${2:-""}"
+  alternate_image="${3:-""}"
```

Comment:
What if users want to issue the command like `kubectl rook-ceph start-debug <deploy> --image=my-image`? Or what if they do `kubectl rook-ceph start-debug --image my-image <deploy>`? This will fail in both cases. We should use the `parse_flags()` function for all flag parsing.

The commented-out `run_status_command()` example should help.

Reply:
Probably parse_flags() also not help in position problem,

We might need to add `getopts` support for that

Reply:
You're right about the position. `parse_flags()` assumes it should stop when it reaches a non-flag argument. We could probably solve that by calling `parse_flags()` twice.

Reply:
THat makes other input more complicated,

How about this input,

`kubectl rook_ceph xyz  --flag1 parmeter1 --flag2 parmater2 --flag3`